metricflow/specs.py

```diff
@@ -269,7 +278,9 @@ def qualified_name(self) -> str:  # noqa: D
         ).qualified_name
 
 
+@dataclass(frozen=True)
 class MeasureSpec(InstanceSpec):  # noqa: D
+    element_name: str
```

Comment:
If you make InstanceSpec a protocol you can then remove it from the signature here. I don't love it but it works better that way.